### PR TITLE
MasterApeAdmin

### PR DESCRIPTION
**Pull request by DeFiFoFum**
The `MasterApeAdmin` contract is an admin helper proxy contract to the MasterApe which adds the following functionality: 

- Add batch MasterApe farms
- Set batch MasterApe farms
- Add "Fixed Percentage Farms": These are farms that need to take a constant percentage of the newly minted BANANA supply. One use is to support "ghost farms" without needing to constantly update allocations. 
- Update/Remove "Fixed Percentage Farms": Change percentage allocations 

When new farms are added or updated through the `MasterApeAdmin` contract, the fixed percentage allocation farms are updated directly after to keep their percentages constant.

### CODE REVIEW

File: contracts/MasterApeAdmin.sol [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    /// @notice Obtain detailed allocation information regarding a MasterApe pool
+    /// @param pid MasterApe pid to pull detailed information from
+    /// @return lpToken Address of the stake token for this pool
+    /// @return poolAllocationPoint Allocation points for this pool
+    /// @return totalAllocationPoints Total allocation points across all pools
+    /// @return poolAllocationPercentMantissa Percentage of pool allocation points to total multiplied by 1e18
+    /// @return poolBananaPerBlock Amount of BANANA given to the pool per block
+    /// @return poolBananaPerDay Amount of BANANA given to the pool per day
+    /// @return poolBananaPerMonth Amount of BANANA given to the pool per month
+    function getDetailedPoolInfo(uint pid) external view returns (
```
Comment (Apegurus) [inline contracts/MasterApeAdmin.sol:124]: Very nice helper function!

File: contracts/MasterApeAdmin.sol [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+    function getTotalAllocationPercent() public view returns (uint256) {
+        return totalFixedPercentFarmPercentage + BASE_PERCENTAGE;
+    }
+
+
+    /** Internal Functions  */
+
+    /// @notice Run through fixed percentage farm allocations and set MasterApe allocations to match the percentage.
+    /// @dev The MasterApe contract manages the BANANA pool percentage on its own which is accounted for in the calculations below.
+    function _syncFixedPercentFarms() internal {
```
Comment (Apegurus) [inline contracts/MasterApeAdmin.sol:328]: Should we add the possibility to send a range to this function? 

Thus not bricking the contract if a gas limit is hit.

**Review by Apegurus — CHANGES_REQUESTED** [review]
Would be nice to add the possibility to send a range/array of PIDs to avoid the contract being bricked by gas limits

File: contracts/MasterApeAdmin.sol (hunk 2 above)
Comment (DeFiFoFum) [reply contracts/MasterApeAdmin.sol:328]: We can find ways to improve the gas efficiency, but the current calculation depends on updating them all at once.

Also, if this function doesn't work, it won't brick the contract as you can still make farm updates without syncing and we can remove fixed percentage farms if there are too many.

Comment (Apegurus) [reply contracts/MasterApeAdmin.sol:328]: Got it!

**Review by Apegurus — APPROVED** [review]

**Pull request merged**